### PR TITLE
Fix QueryProfiler overruns (reset write_trace_iteration)

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Common/QueryProfiler.cpp [hunk 1]
```diff
@@ -120,6 +120,10 @@ QueryProfilerBase<ProfilerImpl>::QueryProfilerBase(const UInt64 thread_id, const
     if (sigaction(pause_signal, &sa, previous_handler))
         throwFromErrno("Failed to setup signal handler for query profiler", ErrorCodes::CANNOT_SET_SIGNAL_HANDLER);
 
+#if defined(OS_LINUX)
+    write_trace_iteration = 0;
```
Comment (alexey-milovidov) [inline src/Common/QueryProfiler.cpp:124]: Does it change anything?

Comment (azat) [reply src/Common/QueryProfiler.cpp:124]: There is some heuristics based on this counter, to not skip profiling completely on overruns, https://github.com/ClickHouse/ClickHouse/blob/e5b811f96d0adc0897f150124b487d6fb46fdc9c/src/Common/QueryProfiler.cpp#L45

Comment (alexey-milovidov) [reply src/Common/QueryProfiler.cpp:124]: Ok but:
- the heuristic is to write every n-th event, does not matter what exactly;
- only some incremental counter is needed, it does not matter what is the initial value.

Comment (azat) [reply src/Common/QueryProfiler.cpp:124]: Yep, it is merely the same, Agree, I will close this.

**Pull request closed without merging**